shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/registry/cache/subscriber/ScalingRegistrySubscriber.java

```diff
@@ -98,6 +108,10 @@ public void scalingTaskFinished(final ScalingTaskFinishedEvent event) {
             log.info("start to delete cache, ruleCacheId={}", ruleCacheId);
             registryCacheManager.deleteCache(SchemaMetaDataNode.getRulePath(event.getTargetSchemaName()), ruleCacheId);
         }
+        if (lockStatus) {
```

Comment:
From code, seems `ScalingTaskFinishedEvent` is not sent in every case, e.g. target DB broken and scaling job failed. The lock might be not released properly.

Reply:
Yes, there are lock release failures when the task is stopped abruptly. Here I will raise a new pull request to solve it

Reply:
OK, then you could comment it in this PR, it's clear to see it later.